src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/MouseEventSender.cs

```diff
@@ -366,7 +366,63 @@ public static void SendMouseEvent(int replaySegment, MouseInputActionData mouseI
+        }
 
+        /// <summary>
+        /// Simulates a mouse event at the screen space position corresponding to the given Transform's world position.
+        /// </summary>
+        /// <param name="mouseTarget">The Transform whose world position will be converted to screen space for the mouse event.</param>
+        /// <param name="leftButton">Simulates left mouse button press if true.</param>
+        /// <param name="middleButton">Simulates middle mouse button press if true.</param>
+        /// <param name="rightButton">Simulates right mouse button press if true.</param>
+        /// <param name="forwardButton">Simulates forward mouse button press if true.</param>
+        /// <param name="backButton">Simulates back mouse button press if true.</param>
+        /// <param name="scroll">Simulates mouse scroll wheel movement.</param>
+        public static void SendMouseEvent(Transform mouseTarget,
+                                          bool leftButton=false,
+                                          bool middleButton=false,
+                                          bool rightButton=false,
+                                          bool forwardButton=false,
+                                          bool backButton=false,
+                                          Vector2 scroll = default)
+        {
+            Vector2 screenPosition = Camera.main.WorldToScreenPoint(mouseTarget.position);
+            #if ENABLE_LEGACY_INPUT_MANAGER
+            SendMouseEventLegacy(screenPosition, Vector2.zero, scroll, leftButton, middleButton, rightButton, forwardButton, backButton);
+            #else
+            var screenWidth = Screen.width;
+            var screenHeight = Screen.height;
+            var normalizedPosition = new Vector2(screenPosition.x / screenWidth, screenPosition.y / screenHeight);
+            SendRawPositionMouseEvent(0, normalizedPosition, leftButton, middleButton, rightButton, forwardButton, backButton, scroll);
+            #endif
+        }
+
+        /// <summary>
+        /// Simulates a mouse event at the specified screen position. Bottom left is (0,0) and top right is (Screen.width, Screen.height).
+        /// </summary>
+        /// <param name="mouseScreenPosition">The screen coordinates where the mouse event will be simulated.</param>
+        /// <param name="leftButton">Simulates left mouse button press if true.</param>
+        /// <param name="middleButton">Simulates middle mouse button press if true.</param>
+        /// <param name="rightButton">Simulates right mouse button press if true.</param>
+        /// <param name="forwardButton">Simulates forward mouse button press if true.</param>
+        /// <param name="backButton">Simulates back mouse button press if true.</param>
+        /// <param name="scroll">Simulates mouse scroll wheel movement.</param>
+        public static void SendMouseEvent(Vector2 mouseScreenPosition,
```

Comment:
This already exists as `SendRawPositionMouseEvent`, let's avoid creating another method for it.

Reply:
For coding practices 100%. The reason why it is just as a repeat method with the same name is to make it easier for the LLM. I wanted it to be 'There is a SendMouseEvent function, you can either pass in a transform or pixel position, either does something sensible.`

@RG-nAmKcAz @vontell On second thought it might be potentially better to create a separate LLMInterface script that can contain both this and KeyboardInput. Then we can just give this public API to runespawn, and it will be much easier to maintain. I will create a new task for the next PR to combine the two if no one objects.

Reply:
We aren't in a rush to release this, so I would like it cleaned up in this PR where the feedback was given instead of deferred to later.  I don't want to introduce new public methods to the SDK that effectively duplicate the functionality of existing methods.

Reply:
@RG-nAmKcAz Okay, I will create the llm interface in this PR.